### PR TITLE
feat(rpc): Setup RPC Crate

### PR DESCRIPTION
**Pull request by refcell**
### Description

Sets up an `rpc` crate.

### Metadata

#7

### CODE REVIEW

**Comment by refcell** [comment]
Going to cut this off here with `unimplemented` for others to impl - I'm not as familiar with how this could work well with reth. Would appreciate any guidance if you have bandwidth @Rjected

File: crates/rpc/src/sequencer.rs [hunk 1]
```diff
@@ -0,0 +1,114 @@
+//! Optimism reth RPC Extension used to forward raw transactions to the sequencer.
```
Comment (merklefruit) [inline crates/rpc/src/sequencer.rs:1]: just curious, why do we need to forward txs to the sequencer from the rollup node?

Comment (refcell) [reply crates/rpc/src/sequencer.rs:1]: This is actually for op-reth - starting to move things out of https://github.com/paradigmxyz/reth/blob/main/crates/optimism/rpc/src/eth/rpc.rs. Happy to nix this if you think it's a better idea to wait on moving op-reth stuff over here

Comment (refcell) [reply crates/rpc/src/sequencer.rs:1]: Actually, think I convinced myself it's worth removing this for now. I'll rework this PR so we can maybe move the RPC across the line.

File: crates/rpc/Cargo.toml [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+jsonrpsee.workspace = true
+async-trait.workspace = true
+
+# `reth` feature flag dependencies
+alloy = { workspace = true, optional = true }
+reqwest = { workspace = true, optional = true }
+thiserror = { workspace = true, optional = true }
+serde_json = { workspace = true, optional = true }
+jsonrpsee-types = { workspace = true, optional = true }
+reth-rpc-eth-types = { workspace = true, optional = true }
```
Comment (mattsse) [inline crates/rpc/Cargo.toml:32]: this can be removed?

File: Cargo.toml [hunk 3]
```diff
@@ -76,7 +77,11 @@ tokio = { version = "1.21", default-features = false }
 
 # Serialization
 serde_json = "1"
-ethereum_ssz = "0.7.1"
+ethereum_ssz = "0.8.0"
+
+# RPC
+jsonrpsee = { version = "0.24", features = ["jsonrpsee-core", "client-core", "server-core", "macros"] }
```
Comment (mattsse) [inline Cargo.toml:83]: these shouldn't be enabled on workspace level and instead on a per crate basis

**Pull request merged**